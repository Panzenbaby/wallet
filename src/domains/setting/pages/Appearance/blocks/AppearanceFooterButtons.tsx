import { Button } from "app/components/Button";
import React from "react";
import { useTranslation } from "react-i18next";
import { useHistory } from "react-router-dom";

interface Properties {
	isSaveDisabled: boolean;
}

export const AppearanceFooterButtons: React.FC<Properties> = ({ isSaveDisabled }: Properties) => {
	const { t } = useTranslation();

	const history = useHistory();

	return (
		<div className="flex justify-end mt-8 space-x-3 w-full">
			<Button data-testid="AppearanceFooterButtons__cancel" variant="secondary" onClick={() => history.go(-1)}>
				{t("COMMON.CANCEL")}
			</Button>

			<Button
				data-testid="AppearanceFooterButtons__save"
				disabled={isSaveDisabled}
				type={isSaveDisabled ? "button" : "submit"}
			>
				{t("COMMON.SAVE")}
			</Button>
		</div>
	);
};
