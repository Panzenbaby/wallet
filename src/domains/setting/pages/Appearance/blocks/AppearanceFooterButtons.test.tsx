import { render, screen } from "@testing-library/react";
import React from "react";

import { AppearanceFooterButtons } from "./AppearanceFooterButtons";

describe("AppearanceFooterButtons", () => {
	it("should render", () => {
		const { asFragment } = render(<AppearanceFooterButtons isSaveDisabled={false} />);

		expect(screen.getByTestId("AppearanceFooterButtons__save")).not.toBeDisabled();

		expect(asFragment()).toMatchSnapshot();
	});

	it("should render disabled", () => {
		const { asFragment } = render(<AppearanceFooterButtons isSaveDisabled={true} />);

		expect(screen.getByTestId("AppearanceFooterButtons__save")).toBeDisabled();

		expect(asFragment()).toMatchSnapshot();
	});
});
