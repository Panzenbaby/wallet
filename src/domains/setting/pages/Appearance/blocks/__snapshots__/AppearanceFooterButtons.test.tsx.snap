// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppearanceFooterButtons should render 1`] = `
<DocumentFragment>
  <div
    class="flex justify-end mt-8 space-x-3 w-full"
  >
    <button
      class="OriginalButton__StyledButton-o0fs2j-0 guPjPB"
      data-testid="AppearanceFooterButtons__cancel"
      type="button"
    >
      <div
        class="flex relative items-center space-x-2"
      >
        Cancel
      </div>
    </button>
    <button
      class="OriginalButton__StyledButton-o0fs2j-0 dABsWA"
      data-testid="AppearanceFooterButtons__save"
      type="submit"
    >
      <div
        class="flex relative items-center space-x-2"
      >
        Save
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`AppearanceFooterButtons should render disabled 1`] = `
<DocumentFragment>
  <div
    class="flex justify-end mt-8 space-x-3 w-full"
  >
    <button
      class="OriginalButton__StyledButton-o0fs2j-0 guPjPB"
      data-testid="AppearanceFooterButtons__cancel"
      type="button"
    >
      <div
        class="flex relative items-center space-x-2"
      >
        Cancel
      </div>
    </button>
    <button
      class="OriginalButton__StyledButton-o0fs2j-0 jmXqYL"
      data-testid="AppearanceFooterButtons__save"
      disabled=""
      type="button"
    >
      <div
        class="flex relative items-center space-x-2"
      >
        Save
      </div>
    </button>
  </div>
</DocumentFragment>
`;
